src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,312 @@
+        let batch_settings = self.batch.into_batch_settings()?;
+
+        let buffer = JsonArrayBuffer::new(batch_settings.size);
+
+        let tls_settings = TlsSettings::from_options(&None)?;
+        let client = HttpClient::new(tls_settings, cx.proxy())?;
+
+        let sink = BatchedHttpSink::new(
+            self.clone(),
+            buffer,
+            request_settings,
+            batch_settings.timeout,
+            client.clone(),
+        )
+        .sink_map_err(|error| error!(message = "Fatal AppSignal sink error.", %error));
+
+        let healthcheck = healthcheck(
+            endpoint_uri(&self.endpoint, "vector/healthcheck")?,
+            push_api_key,
+            client,
+        )
+        .boxed();
+
+        Ok((super::VectorSink::from_event_sink(sink), healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        Input::new(DataType::Metric | DataType::Log)
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+
+pub struct AppsignalEventEncoder {
+    transformer: Transformer,
+}
+
+impl HttpEventEncoder<serde_json::Value> for AppsignalEventEncoder {
+    fn encode_event(&mut self, mut event: Event) -> Option<serde_json::Value> {
+        self.transformer.transform(&mut event);
+
+        match event {
+            Event::Log(log) => Some(json!({ "log": log })),
+            Event::Metric(metric) => Some(json!({ "metric": metric })),
+            _ => panic!("The AppSignal sink does not support this type of event: {event:?}"),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl HttpSink for AppsignalSinkConfig {
+    type Input = serde_json::Value;
+    type Output = Vec<BoxedRawValue>;
+    type Encoder = AppsignalEventEncoder;
+
+    fn build_encoder(&self) -> Self::Encoder {
+        AppsignalEventEncoder {
+            transformer: self.encoding.clone(),
+        }
+    }
+
+    async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Bytes>> {
+        let uri = endpoint_uri(&self.endpoint, "vector/events")?;
+        let mut request = Request::post(uri).header(
+            AUTHORIZATION,
+            format!("Bearer {}", self.push_api_key.inner()),
+        );
+
+        let mut body = crate::serde::json::to_bytes(&events)?.freeze();
+
+        let compression = match self.compression {
+            AppsignalCompression::Gzip => {
+                request = request.header("Content-Encoding", "gzip");
+                Compression::gzip_default()
+            }
+            AppsignalCompression::Zlib => {
+                request = request.header("Content-Encoding", "deflate");
+                Compression::zlib_default()
+            }
+        };
+        let mut compressor = Compressor::from(compression);
+        write_all(&mut compressor, 0, &body)?;
+        body = compressor.finish().context(CompressionFailedSnafu)?.into();
+        request.body(body).map_err(Into::into)
+    }
+}
+
+pub(crate) async fn healthcheck(
```

Comment:
`pub` items need a doc comment.

Reply:
It doesn't have to be public, so I'll remove the `pub(crate)` part.